drud/ddev
# [docs] Add consolidated Architecture page

## The Problem/Issue/Bug:

The [Guides](https://ddev.readthedocs.io/en/latest/users/topics/phpstorm/) section is fairly thin, and its helpful [What’s in the .ddev Directory?](https://ddev.readthedocs.io/en/latest/users/topics/directory-tour/) page would go great with the existing architectural overview.

## How this PR Solves The Problem:

This adds a cheeky introduction to the architecture page that tells us what the page is for and more broadly describes _what DDEV is_ before taking you on a tour of the directories it uses and then its under-the-hood container layout.

It moves a little bit of content into the config and command references so the directory tour can focus on paths and files, and of course adds links to those references for learning and exploration.

It also gently re-titles “Things Not to Look at or Mess With” to benefit the reader’s dignity and healthy sense of curiosity, since we’re intentionally looking at those files here and learning why we shouldn’t mess with them. (The reader could go on to mess with them, we just want them to understand why they’d be wise not to!)

It requires the diligent author to add two redirects:

1. `/users/topics/directory-tour/` → `/users/basics/architecture/`
2. `/users/topics/how-ddev-works/` → `/users/basics/architecture/`

The page title doesn’t change, so search performance should be roughly the same—with the benefit of potential hits on “architecture”.

Moving the directory tour closer to introductory usage probably makes a lot of sense; you’ll want to know what all those new local files are, probably more than you’ll need to understand the container layout that makes them do stuff.

## Manual Testing Instructions:

[Examine this branch locally](https://ddev.readthedocs.io/en/latest/developers/testing-docs/#preview-changes) or check out the [automatic preview build](https://ddev--4379.org.readthedocs.build/en/4379/users/basics/architecture/).


## Automated Testing Overview:

n/a

## Related Issue Link(s):

#4188

## Release/Deployment notes:

n/a


<a href="https://gitpod.io/#https://github.com/drud/ddev/pull/4379"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
